src/LiveComponent/assets/src/Component/plugins/QueryStringPlugin.ts

```diff
@@ -0,0 +1,52 @@
+import Component from '../index';
+import { PluginInterface } from './PluginInterface';
+import {
+    setQueryParam, removeQueryParam,
+} from '../../url_utils';
+
+type QueryMapping  = {
+    name: string,
+}
+
+export default class implements PluginInterface {
+    private element: Element;
+    private mapping: Map<string,QueryMapping> = new Map;
+
+    attachToComponent(component: Component): void {
+        this.element = component.element;
+        this.registerBindings();
+
+        component.on('connect', (component: Component) => {
+            this.updateUrl(component);
```

Comment:
I wasn't sure about this one, Livewire illustrates the `keep` behavior with an initial empty value, and I also considered a non-empty initial value, that could be set on component mount for example, based on some context. But it may be too much for the moment, so let's wait for the `keep` implementation!